### PR TITLE
plugins/nvim-ufo: migrate to mkNeovimPlugin

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/2362

### CODE REVIEW

File: plugins/by-name/nvim-ufo/default.nix [hunk 1]
```diff
@@ -1,87 +1,134 @@
...
 let
-  cfg = config.plugins.nvim-ufo;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options.plugins.nvim-ufo = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "nvim-ufo";
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "nvim-ufo";
+  originalName = "nvim-ufo";
```
Comment (GaetanLepage) [inline plugins/by-name/nvim-ufo/default.nix:10]:
```suggestion
```
I think that this defaults to `name`.

File: plugins/by-name/nvim-ufo/default.nix [hunk 2]
```diff
@@ -1,87 +1,133 @@
...
+      "maxheight"
+    ]
+    [
+      "preview"
+      "mappings"
+    ]
+  ];
+
+  settingsOptions = {
+    open_fold_hl_timeout = defaultNullOpts.mkInt 400 ''
```
Comment (GaetanLepage) [inline plugins/by-name/nvim-ufo/default.nix:50]:
```suggestion
    open_fold_hl_timeout = defaultNullOpts.mkUnsignedInt 400 ''
```

File: plugins/by-name/nvim-ufo/default.nix [hunk 3]
```diff
@@ -1,87 +1,133 @@
...
-    };
+  maintainers = [ lib.maintainers.khaneliman ];
 
-    openFoldHlTimeout = helpers.defaultNullOpts.mkInt 400 ''
+  # TODO: added 2024-10-05 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "openFoldHlTimeout"
+    "providerSelector"
+    "closeFoldKinds"
```
Comment (GaetanLepage) [inline plugins/by-name/nvim-ufo/default.nix:20]: Isn't the new name `close_fold_kinds_for_ft` ? If so, you need to manually add an `mkRenamedOptionSubmodule` to `imports`.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: plugins/by-name/nvim-ufo/default.nix (hunk 3 above)
Comment (khaneliman) [reply plugins/by-name/nvim-ufo/default.nix:20]: It is deprecated and not removed. They are different types, as well. This is a rename for compatibility but the documented option is the new name.

Comment (khaneliman) [reply plugins/by-name/nvim-ufo/default.nix:20]: https://github.com/kevinhwang91/nvim-ufo/commit/7b14dd650b57291a49f037f1ede325d46e08b258

File: plugins/by-name/nvim-ufo/default.nix (hunk 2 above)
Comment (khaneliman) [reply plugins/by-name/nvim-ufo/default.nix:50]: Wondering if the winblend and maxheight should be, as well..

Comment (khaneliman) [reply plugins/by-name/nvim-ufo/default.nix]: Updated them both, winblend is 0-100 and maxheight doesn't sound like something that could be negative.

File: plugins/by-name/nvim-ufo/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply plugins/by-name/nvim-ufo/default.nix:20]: Ok I see. So we do want the values given to `closeFoldKinds` to be forwarded to `settings.close_fold_kinds` then, and let the plugin warn the user about the deprecation. I have correctly understood ?

If so, your implem looks good.

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by khaneliman** [comment]
@mergify queue

**Pull request merged**